lib/src/base_request.dart

```diff
@@ -88,8 +88,18 @@ abstract class BaseRequest {
   bool get finalized => _finalized;
   bool _finalized = false;
 
-  BaseRequest(this.method, this.url)
-      : headers = LinkedHashMap(
+  BaseRequest(String method, this.url)
+      : method = [
+          'HEAD',
+          'GET',
+          'POST',
+          'PUT',
+          'PATCH',
+          'DELETE',
+        ].contains(method)
```

Comment:
Where does this list come from?

The [HTTP RFC](https://tools.ietf.org/html/rfc7231#section-4.3) specifies:
 * GET 
 * HEAD
 * POST
 * PUT
 * DELETE
 * CONNECT
 * OPTIONS
 * TRACE

as defined standard methods, and we are missing a couple of these. The IANA [method registry](https://www.iana.org/assignments/http-methods/http-methods.xhtml) is much larger.

The only real requirement is that it's a ["token"](https://tools.ietf.org/html/rfc7230#section-3.2.6), to allowing servers to support custom methods. so maybe just check that.
Something like:
```dart
static final _tokenRE = RegExp(r"^[\w!#%&'*+\-.^`|~]+$");
static String _validateMethod(String method) {
  if (!_tokenRE.hasMatch(method)) {
    throw ArgumentError.value(method, "method", "Not a valid method");
  }
  return method;
}
...
BaseRequest(String method, this.url) : method = _validateMethod(method);
```

This should still ensure that the method is parsed as a single token by the receiver, but not put any unnecessary restrictions on it.
(I'm using a `RegExp` here, if performance is an issue, it can probably be optimized).

Reply:
If I understand correctly, all methods listed in the IANA registry can be matched just by `^[\w\-]+$`.

I've committed: `"^[\w!#%&'*+\-.^|~]+$"`, but can I ask why characters like `#%&...` are expected to be considered as valid?